html/user/openssl_sign_action.php

```diff
@@ -0,0 +1,134 @@
+<?php
+// This file is part of BOINC.
+// http://boinc.berkeley.edu
+// Copyright (C) 2008 University of California
+//
+// BOINC is free software; you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// BOINC is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+// See the GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+
+require_once("../inc/boinc_db.inc");
+require_once("../inc/user.inc");
+require_once("../inc/util.inc");
+
+// Needs to be defined, unknown use.
+check_get_args(array("tnow", "ttok"));
+
+// Check the user is online
+$user = get_logged_in_user();
+check_tokens($user->authenticator);
+
+// For some reason, I'm only able to use numbers, no letters.
+// Need to provide security, but full functionality is likewise desirable.
+$user_data = sanitize_tags(post_str("user_data", true));
```

Comment:
You don't have to sanitize it if the string is going to be only hashed. These crypto algs can deal with any byte-sequence. (By design, that is. If they can't, then OpenSSL is total shit, which i hope is not.)

Reply:
True, it's not getting stored in SQL nor is it being stored anywhere in the website, I left this in because it's standard for pretty much every other data input & I was unsure if you could potentially maliciously escape the code without it. If it's safe to remove then we could go ahead and do so.

Reply:
I am worried that it is not clear what this two functions do. What chars are safe? External systems may need to duplicate this sanitation in order to check the signature.